HLTrigger/Timer/src/FastTimerService.cc

```diff
@@ -38,1425 +39,1365 @@
-    for (auto const & name: tns.getTrigPaths()) {
-      out << "FastReport              "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_active        / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_premodules    / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_intermodules  / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_postmodules   / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_overhead      / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_total         / (double) summary.count << "  "
-          << name << '\n';
-    }
-    out << '\n';
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active       Pre-     Inter-  Post-mods   Overhead      Total  EndPath" << '\n';
-    for (auto const & name: tns.getEndPaths()) {
-      out << "FastReport              "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_active        / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_premodules    / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_intermodules  / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_postmodules   / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_overhead      / (double) summary.count << " "
-          << std::right << std::setw(10) << m_stream.paths[pid][name].summary_total         / (double) summary.count << "  "
-          << name << '\n';
-    }
-  }
-  out << '\n';
-  if (m_enable_timing_modules) {
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    for (auto & keyval: m_stream.modules) {
-      std::string const & label  = keyval.first;
-      ModuleInfo  const & module = keyval.second;
-      out << "FastReport              " << std::right << std::setw(10) << module.summary_active  / (double) summary.count << "  " << label << '\n';
-    }
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    out << '\n';
-    //out << "FastReport " << (m_use_realtime ? "(real time) " : "(CPU time)  ")    << "     Active  Module" << '\n';
-    for (auto & keyval: m_stream.moduletypes) {
-      std::string const & label  = keyval.first;
-      ModuleInfo  const & module = keyval.second;
-      out << "FastReport              " << std::right << std::setw(10) << module.summary_active  / (dou
```

Comment:
I think this is a race condition. Since mulitple SubProcesses could reach `postEvent` at the same time and then try to update `stream.total` simultaneously.